node/src/main/scala/coop/rchain/node/api/json/JsonSchemaDerivation.scala

```diff
@@ -0,0 +1,81 @@
+import coop.rchain.casper.PrettyPrinter
+import coop.rchain.casper.protocol.{
+  BlockInfo,
+  BondInfo,
+  DeployData,
+  DeployInfo,
+  JustificationInfo,
+  LightBlockInfo,
+  RejectedDeployInfo
+}
+import coop.rchain.models.Par
+import coop.rchain.models.syntax._
+import coop.rchain.node.api.WebApi._
+import coop.rchain.node.web.TransactionInfo
+import endpoints4s.{Invalid, Valid}
+
+/**
+  * JsonSchema derivations for Web API request/response objects (Scala sealed traits and case classes)
+  */
+trait JsonSchemaDerivations extends JsonSchemaDerivationsBase {
+
+  // format: off
+
+  implicit lazy val deployDataSchema      : JsonSchema[DeployData]                   = schemaRecord
+  implicit lazy val deployRequestSchema   : JsonSchema[DeployRequest]                = schemaRecord
+  implicit lazy val versionInfoSchema     : JsonSchema[VersionInfo]                  = schemaRecord
+  implicit lazy val apiStatusSchema       : JsonSchema[ApiStatus]                    = schemaRecord
+  implicit lazy val exploreDeployReqSchema: JsonSchema[ExploreDeployRequest]         = schemaRecord
+  implicit lazy val dataAtNameReqSchema   : JsonSchema[DataAtNameByBlockHashRequest] = schemaRecord
+  implicit lazy val rhoResponseSchema     : JsonSchema[RhoDataResponse]              = schemaRecord
+  implicit lazy val bondInfoSchema        : JsonSchema[BondInfo]                     = schemaRecord
+  implicit lazy val justInfoSchema        : JsonSchema[JustificationInfo]            = schemaRecord
+  implicit lazy val rejectedInfoSchema    : JsonSchema[RejectedDeployInfo]           = schemaRecord
+  implicit lazy val lightBlockInfoSchema  : JsonSchema[LightBlockInfo]               = schemaRecord
+  implicit lazy val deployInfoSchema      : JsonSchema[DeployInfo]                   = schemaRecord
+  implicit lazy val blockInfoSchema       : JsonSchema[BlockInfo]                    = schemaRecord
+//  implicit lazy val transactionInfoSchema : JsonSchema[TransactionInfo]              = schemaRecord
+
+  // Web API Rholang types (subset of protobuf generated types)
+  implicit lazy val rhoExprSchema: JsonSchema[RhoExpr] =
+    lazySchema("RhoExprRef")(schemaTagged[RhoExpr]).withDescription("Rholang expression (Par type)")
+  // TODO: add ExprXXX types (not necessary for derivation but for short type name without a namespace)
+  implicit lazy val rhoUnforgSchema     : JsonSchema[RhoUnforg]      = schemaTagged
+  implicit lazy val unforgPrivateSchema : JsonSchema[UnforgPrivate]  = schemaRecord
+  implicit lazy val unforgDeploySchema  : JsonSchema[UnforgDeploy]   = schemaRecord
+  implicit lazy val unforgDeployerSchema: JsonSchema[UnforgDeployer] = schemaRecord
+
+  // Protobuf generated Rholang types (from models project)
+//  implicit lazy val parSchema: JsonSchema[Par] = lazySchema("ParRef", schemaRecord[Par])
+  /** TODO: add all referenced types in Par type, see [[coop.rchain.node.encode.JsonEncoder]] as example for circe derivations */
```

Comment:
@stanislavlyalin Here are missing types that needs to be defined so that the whole Web API can be implemented.

Reply:
@tgrospic, can you explain, please. Do you mean JSON serialization/deserialization implicit vals for messages used in `Par`? E.g. `Send`, `Receive`, `New`... `JsonEncoder.scala` already has it.

Reply:
Yes, we need definitions for JsonSchema for all types used in `Par`. JsonEncoder can be an example what types are necessary to define but definitions there cannot be reused for this version of API because circe encoders/decorders are generated from JsonSchema and have different format.